File: src/main/java/com/ris/volunteerplatform/entity/ApplicationStatus.java
package com.ris.volunteerplatform.entity;

/** Статус заявки волонтёра на участие в задаче. */
public enum ApplicationStatus {
    PENDING,
    APPROVED,
    REJECTED,
    WITHDRAWN,
    /** Организатор закрыл заявку: задача выполнена успешно */
    COMPLETED_SUCCESS,
    /** Организатор закрыл заявку: задача не была выполнена */
    COMPLETED_FAILURE
}

File: src/main/ui/src/utils/applicationStatus.ts
import type { ApplicationStatus } from '../types';

export const APPLICATION_STATUS_RU: Record<ApplicationStatus, string> = {
  PENDING: 'На рассмотрении',
  APPROVED: 'Одобрена',
  REJECTED: 'Отклонена',
  WITHDRAWN: 'Отозвана',
  COMPLETED_SUCCESS: 'Завершена (выполнено)',
  COMPLETED_FAILURE: 'Завершена (не выполнено)',
};

export function applicationStatusRu(status: ApplicationStatus | string): string {
  if (Object.prototype.hasOwnProperty.call(APPLICATION_STATUS_RU, status)) {
    return APPLICATION_STATUS_RU[status as ApplicationStatus];
  }
  return String(status);
}
